sysdig/data_source_sysdig_secure_tenant_external_id.go

```diff
@@ -0,0 +1,49 @@
+package sysdig
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceSysdigSecureTenantExternalID() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		ReadContext: dataSourceSysdigSecureTenantExternalIDRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(timeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"external_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// Retrieves the information of a resource form the file and loads it in Terraform
+func dataSourceSysdigSecureTenantExternalIDRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudAccountClient(meta.(SysdigClients))
```

Comment:
This makes these APIs via the cloud_account (i.e cloudsec client). If we plan to deprecate the client, we will need to move these API calls to the cloudauth based client.

Reply:
good callout. I hadn't noticed this.

Reply:
it's a good callout, but these "clients" aren't actually different things. It's just a means to organize code along organizational boundaries.  Let me see if I can organize this in a better way.

Reply:
I've created an onboarding "client", that consolidates things a bit.